ds3-autogen-go/src/main/kotlin/com/spectralogic/ds3autogen/go/generators/request/RequestGeneratorUtil.kt

```diff
@@ -272,29 +270,26 @@ fun toWithConstructor(ds3Param: Ds3Param): WithConstructor {
 /**
  * Determines if a Ds3Request's Resource describes a required argument or not.
  */
-fun isGoResourceAnArg(
-        resource: Resource?,
-        includeIdInPath: Boolean): Boolean {
-    return resource != null && includeIdInPath
+fun Ds3Request.isGoResourceAnArg(): Boolean {
+    return this.resource != null && this.includeInPath
 }
 
 /**
  * Creates an argument from a resource. Notification resource args are simplified to notificationId.
  * If the resource does not describe a valid argument, such as a singleton, an error is thrown.
- * @param resource A Ds3Request's Resource
  */
-fun getGoArgFromResource(resource: Resource?): Arguments {
-    if (isResourceSingleton(resource)) {
-        throw IllegalArgumentException("Cannot create an argument from a singleton resource: " + resource.toString())
+fun Ds3Request.getGoArgFromResource(): Arguments {
+    if (isResourceSingleton(this.resource)) {
```

Comment:
pulling this out into a local variable may help clean this up a bit so that you don't have to type `this` everywhere.